mqtt/mqtt-bridge/src/token_source.rs

```diff
@@ -13,7 +15,7 @@ define_encode_set! {
 
 #[async_trait]
 pub trait TokenSource {
-    async fn get(&self, expiry: &DateTime<Utc>) -> Result<String, TokenSourceError>;
+    async fn get(&self, expiry: &DateTime<Utc>) -> Result<Option<String>, Error>;
```

Comment:
I don't think we need to use std::io::Error for our business code errors. It's better we use our own enums

Reply:
I changed because it was needed for try_join and it's not exposed from the client